src/libraries/System.Private.CoreLib/src/System/Threading/Wasi/WasiPollWorld.wit.imports.wasi.io.v0_2_0.PollInterop.cs

```diff
@@ -0,0 +1,56 @@
+// Generated by `wit-bindgen` 0.27.0. DO NOT EDIT!
+// <auto-generated />
+#nullable enable
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Collections;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+
+namespace WasiPollWorld.wit.imports.wasi.io.v0_2_0
+{
+    internal static class PollInterop {
+
+        internal static class PollWasmInterop
+        {
+            [DllImport("wasi:io/poll@0.2.0", EntryPoint = "poll"), WasmImportLinkage]
+            internal static extern void wasmImportPoll(nint p0, int p1, nint p2);
+
+        }
+
+        internal  static unsafe uint[] Poll(List<global::WasiPollWorld.wit.imports.wasi.io.v0_2_0.IPoll.Pollable> @in)
+        {
+
+            byte[] buffer = new byte[4 * @in.Count];
+            var gcHandle = GCHandle.Alloc(buffer, GCHandleType.Pinned);
+            var address = gcHandle.AddrOfPinnedObject();
+
+            for (int index = 0; index < @in.Count; ++index) {
+                global::WasiPollWorld.wit.imports.wasi.io.v0_2_0.IPoll.Pollable element = @in[index];
+                int basePtr = (int)address + (index * 4);
+                var handle = element.Handle;
+                BitConverter.TryWriteBytes(new Span<byte>((void*)(basePtr + 0), 4), unchecked((int)handle));
+
+            }
+
+            var retArea = new uint[2];
+            fixed (uint* retAreaByte0 = &retArea[0])
+            {
+                var ptr = (nint)retAreaByte0;
+                PollWasmInterop.wasmImportPoll((int)address, @in.Count, ptr);
+
+                var array = new uint[BitConverter.ToInt32(new Span<byte>((void*)(ptr + 4), 4))];
+                new Span<uint>((void*)(BitConverter.ToInt32(new Span<byte>((void*)(ptr + 0), 4))), BitConverter.ToInt32(new Span<byte>((void*)(ptr + 4), 4))).CopyTo(new Span<uint>(array));
```

Comment:
None of these `BitConverter` calls are needed. We can just use the `retArea` directly. This pattern should be removed from the entire PR.

Reply:
Keep in mind this is all generated code, so it can be difficult to avoid all redundancy in general.  I'll see what I can do in `wit-bindgen` to address this particular case, but given the tradeoffs between code generator complexity and generated code cleanliness, the output will likely never be ideal.

Reply:
Oh, I didn't realize that. Thanks for clarifying that.

Reply:
Also I will merge the code in current form upstream in few moments. 
We will iterate on it as the wit-bindgen improves in next PRs. 

I opened the conversation about the generated code here, because it was convenient and all the right people are here.